HashRegistrarSimplified.sol

```diff
@@ -0,0 +1,367 @@
+
+
+
+/*
+
+Temporary Hash Registrar 
+========================
+
+This is a simplified version of a hash registrar. It is purporsefully limited:
+names cannot be six letters or shorter, new auctions will stop after 4 years
+and all ether still locked after 8 years will become unreachable.
+
+The plan is to test the basic features and then move to a new contract in at most
+2 years, when some sort of renewal mechanism will be enabled.
+
+*/
+
+import 'ENS.sol';
+
+
+contract Deed {
+    /* 
+    The Deed is a contract intended simply to hold ether
+    It can be controlled only by the registrar and can only send ether back to the owner.
+    */
+    Registrar public registrar;
+    address constant burn = 0xdead;
+    uint public creationDate;
+    address public owner;
+    event OwnerChanged(address newOwner);
+    event DeedClosed();
+    bool active;
+
+    modifier onlyRegistrar {
+        if (msg.sender != address(registrar)) throw;
+        _
+    }
+
+    modifier onlyActive {
+        if (!active) throw;
+        _
+    }
+    
+    function Deed() {
+        registrar = Registrar(msg.sender);
+        creationDate = now;
+        active = true;
+    }
+        
+    function setOwner(address newOwner) onlyRegistrar {
+        owner = newOwner;
+        OwnerChanged(newOwner);
+    }
+    
+    function setBalance(uint newValue) onlyRegistrar onlyActive {
+        // Check if it has enough balance to set the value
+        if (this.balance < newValue) throw;
+        // Send the difference to the owner
+        if (!owner.send(this.balance - newValue)) throw;
+    }
+
+    function closeDeed(uint refundRatio) onlyRegistrar onlyActive {
+        active = false;            
+        if (! burn.send(((1000 - refundRatio) * this.balance)/1000)) throw;
+        DeedClosed();
+        destroyDeed();
+    }    
+
+    function destroyDeed() {
+        if (active) throw;
+        if(owner.send(this.balance)) 
+            selfdestruct(burn);
+    }
+
+    /* The default function just receives an amount */
+    function () {}
+}
+
+contract Registrar {
+    ENS public ens;
+    bytes32 public rootNode;
+
+    mapping (bytes32 => entry) public entries;
+    mapping (bytes32 => Deed) public sealedBids;
+    
+    enum Mode { Open, Auction, Owned, Forbidden }
+    uint32 constant auctionLength = 7 days;
+    uint32 constant revealPeriod = 24 hours;
+    uint constant minPrice = 0.01 ether;
```

Comment:
Mostly just that it would exert gradually increasing pressure to do something. But then, a rising price of ether is likely to do that too. Perhaps we could even do away with the hard deadline altogether?